lint/rule.go

```
@@ -55,11 +55,13 @@ func (r *FileRule) GetName() RuleName {
 	return r.Name
 }
 
-// Lint forwards the FileDescriptor to the LintFile method defined on the
-// FileRule.
-func (r *FileRule) Lint(fd *desc.FileDescriptor) []Problem {
-	if r.OnlyIf == nil || r.OnlyIf(fd) {
-		return r.LintFile(fd)
+// Lint accepts a Descriptor and applies LintFile on it
+// only if the descriptor is a FileDescriptor.
+func (r *FileRule) Lint(d desc.Descriptor) []Problem {
+	if f, ok := d.(*desc.FileDescriptor); ok {
```

We do not want to do this. This is a great way to get silent failures.

---

Nevermind. I understand the PR better now.